### DISCUSSION TITLE
Shiny.Push (iOS) - OnEntry and OnReceived placements in Apple PushManager

### DISCUSSION THREAD

**Discussion by MGohil** [shinyorg/shiny#1312] [Q&A]
Hi @aritchie :

I am using Shiny.Push with Azure Push Hub. 

![image](https://github.com/shinyorg/shiny/assets/16605284/b56d5948-98df-4494-a378-a2af7a9556f4)

While testing in iOS I noticed and found that,
**OnEntry** is called when push is coming and App is active/foreground and I am not even Tapping the notification. In this case **OnReceived** should be called as per above documentation.

I believe, following method invokes immediately when Push arrives and app is active/foreground. This is calling TryProcessingIncomingNotification, which in turn calls OnEntry method. I think this should call OnReceived instead of OnEntry.
```
public void OnWillPresentNotification(UNNotification notification, Action<UNNotificationPresentationOptions> completionHandler)
{
    this.TryProcessIncomingNotification(notification, "Foreground remote notification received", () =>
        completionHandler.Invoke(UNNotificationPresentationOptions.List | UNNotificationPresentationOptions.Banner)
    );
}
```

Also, following Handle method is called when app is launched. If it is launched via Push notification tap, It should then call OnEntry and not OnReceived. 
```
public void Handle(UIApplicationLaunchEventArgs args)
{
    if (args.RemoteNotifications == null)
        return;

       ............
       
    this.services.RunDelegates<IPushDelegate>(
        x => x.OnReceived(push),
        this.logger
    );
}
```

Are these two instances, an intended behaviour or actually an issue? 

Thanks.

**Comment by aritchie** [comment]
File a PR if you've analyzed the bug to this extent please or file an appropriate issue.

Please quit tagging me on the forums.  This is meant for community questions, not technical support.
